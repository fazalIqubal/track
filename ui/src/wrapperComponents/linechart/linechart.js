import React from 'react';
import ReactApexChart from 'react-apexcharts';

export class LineChart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      options: {
        chart: {
          height: 350,
          type: 'line',
          zoom: {
            enabled: false
          }
        },
        dataLabels: {
          enabled: false
        },
        chart: {
          toolbar: {
            show: false
          }
        },
        stroke: {
          curve: 'straight'
        },
        title: {
          text: '',
          align: 'left'
        },
        grid: {
          row: {
            colors: [], // takes an array which will be repeated on columns
            opacity: 0.5
          },
        },
        xaxis: {
          categories: [],
        },
        noData: {
          text: "No Onboarding Statistics Available For Tenants",
          align: 'center',
          verticalAlign: 'middle',
          offsetX:  0,
          offsetY: 0
        }
      },
      series: [],
    }
  }

  componentDidMount(){
    let {categories, titleText, colors } = this.props
    let { options } = this.state
    if(titleText){
      options.xaxis.categories = categories || []
      options.title.text = titleText
      options.grid.row.colors= colors
      this.setState({ options})
    }
  }

  render() {
    return (
      <div className="lineChart">
        {this.state.options.title.text&&
          <ReactApexChart
          options={this.state.options}
          series={this.props.series} 
          height={350}
          type='line'
        />
        }
    </div> 
    )
  }
}

