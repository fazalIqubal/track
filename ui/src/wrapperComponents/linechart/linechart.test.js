import React from 'react';
import {LineChart} from './linechart';
import ReactDOM from 'react';
import {configure, shallow} from 'enzyme';
import { shallowToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';


configure({adapter: new Adapter()});

describe('should render correctly', () =>{
    let lineChartOutput;
    beforeEach(() => {
        lineChartOutput= shallow(<LineChart/>)
    });
    afterEach(()=>{
        lineChartOutput= null
    })
    it('should renders correctly', () => {
        expect(lineChartOutput).toMatchSnapshot();
       });

    it('should containe a linechart component', () =>{
        expect(lineChartOutput.find('.lineChart').exists()).toEqual(true)
    })
})


