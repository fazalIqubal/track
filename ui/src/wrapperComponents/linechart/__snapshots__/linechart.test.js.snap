// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly should render correctly  1`] = `null`;

exports[`should render correctly should renders correctly 1`] = `ShallowWrapper {}`;
